### ISSUE TITLE
Alignment failure

### ISSUE THREAD

**Issue by bgottula** [seeing-things/track#157]
On 5 Jan 2020, alignment failed. The `least_squares` function produced a solution but the cost function value of the solution was very high. It is unclear why. I re-ran align.py a second time with no changes and it produced nearly the same result, so the problem was repeatable under the same conditions and using the same code.

A few possible causes to be ruled out:

- [ ] Plate solver failure
- [x] Bad transformation from equatorial coordinates in plate solver result to topocentric coordinates
- [x] `scipy.optimize.least_squares` produced a bad result with otherwise good code and observables, perhaps due to finding a local minimum or perhaps the shape of the cost function does not match its assumptions
- [x] Bug in coordinate transformations used in residual calculations
- [x] Deficiency in mount model (a math problem as opposed to an implementation error)

**Comment by bgottula** [comment]
Using the same set of observations, I dropped one observation at a time and called `solve_model` with the remaining set to determine if perhaps one of the observations was faulty (perhaps the plate solver failed). However none of these runs produced a good result. If plate solver failures are to blame there must have been more than one such failure.

**Comment by bgottula** [comment]
To rule out issues with `least_squares` I used `scipy.optimize.brute` on the same set of observations. Since `brute` is painfully slow I used a recursive approach where on each run I narrowed the range for each parameter around the lowest-cost value found in the previous run. After four iterations it was clear that `brute` was converging on the same solution as `least_squares`. The only difference was that the `pole_rot_axis_az` parameter was shifted by 180-degrees away and `pole_rot_angle` was negated, but this is just an aliased version of the same solution. This experiment seems to rule out `least_squares` as the limiting factor in the quality of the solution. The only other possible explanation relating to the use of these optimizing functions is if my initial set of bounds for each parameter is too restrictive, however I am quite confident that this cannot be the case; each parameter is only bounded to [-180, +180] degrees which should already be more permissive than is actually strictly necessary.

**Comment by bgottula** [comment]
For good measure, I ran `least_squares` with bounds expanded to [-360, +360] for each parameter. This did not change the result.

**Comment by bgottula** [comment]
Tried another variation of outlier rejection by running the `least_squares` solver on only observations in the northern half of the sky (corresponding to the "west" side of the mount meridian) and then with only observations in the southern half of the sky. These both also failed.

Still can't rule out plate solving issues, but if the plate solver is to blame then it must have generated several bad results scattered all over the sky.

**Comment by bgottula** [comment]
I think I've exhausted what I can reasonably do with the existing observation data. I made a few edits to align.py such that it also saves the camera frames to disk whenever the plate solver succeeds so that further debugging is possible. It also stores the equatorial coordinates found by the plate solver in addition to the topocentric coordinates found after transforming them. Further debugging will need to wait until I'm able to attempt alignment again.

**Comment by bgottula** [comment]
I performed multiple alignments this evening and had no failures. I did find that by modeling the tilt of the camera I could greatly improve the RMS pointing error but even without this I was always able to achieve reasonable pointing error. I suppose this case will have to remain a mystery. Fortunately the code has been adjusted such that all of the observables including the camera images are saved to disk so if this occurs again we should have enough data to root cause it properly.
